tutorials/tts/NeMo_TTS_Primer.ipynb

```diff
@@ -0,0 +1,1736 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "FxsyNfSBRKqc",
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Duration Prediction Models\n",
+    "\n",
+    "There are some significant weaknesses to auto-regressive systems. Most notably:\n",
+    "\n",
+    "*   The attention is not robust, causing them to sometimes repeat words, skip words, or generate non-sensical output.\n",
+    "*   Because spectrograms are long (100s-1000s of frames) generating them 1 frame at a time makes training & inference slow.\n",
+    "*   The user has little to no control over how the sentence is spoken.\n",
+    "\n",
+    "To counteract these weaknesses, it is becoming more common for TTS models to use **explicit duration prediction**.\n",
+    "\n",
+    "Duration prediction means that the model encoder predicts the duration (ie. number of spectrogram frames) of each character directly (instead of the model determining the character lengths internally and then predicitng a stop token to end the predicted sequence).\n",
+    "\n",
+    "Most model architectures that support duration prediction, such as [FastPitch](https://fastpitch.github.io/), are transformer based.\n",
+    "\n",
+    "Lets take look at FastPitch:\n",
+    "\n",
+    "<div>\n",
+    "<img src=\"https://raw.githubusercontent.com/NVIDIA/NeMo/tts_primer/tutorials/tts/images/fastpitch_diagram.png\" width=\"600\", height=\"500\"/>\n",
+    "</div>\n",
+    "\n",
+    "At a high level it contains:\n",
+    "\n",
+    "1.   An encoder consisting of a **feed-forward transformer block (FFTr)**, which transforms the input character embeddings into character pronounciation embeddings.\n",
+    "2.   A temporal CNN which takes the encoder output and predicts the duration & pitch of each character. At training time, the ground truth pitch & duration information are fed to the model (similar to teacher forcing).\n",
+    "3.   Each encoder output is repeated a number of times equal to the predicted duration. The repeated encoder output is the same length as the final spectrogram. For example, for characters 'abc' and predicted durations (2, 3, 1) we get the encoder output repeated 'aabbbc'. \n",
+    "4.   A decoder consiting of a FFTr that transforms this encoder output into the predicted spectrogram.\n",
+    "\n",
+    "Some advantages to this approach are:\n",
+    "\n",
+    "*   The duration prediction makes the output consistent and robust.\n",
+    "*   The model if fast because the entire text & spectrogram sequence can be processed in parallel.\n",
+    "*   You can customize the pronunciation by manually selecting the pitch & duration of each character or word.\n",
```

Comment:
I think to be precise we should avoid "pronunciation" in this context, since that might get confusing with the phonemes/heteronyms section earlier. Maybe "prosody"?